classes/app.rb

```diff
@@ -0,0 +1,164 @@
+require_relative './book'
+require_relative './classroom'
+require_relative './person'
+require_relative './rentals'
+require_relative './student'
+require_relative './teacher'
+require 'date'
+
+class App
+  def initialize
+    @rentals = []
+    @people = []
+    @books = []
+  end
+
+  def console_entry_point
+    puts ''
+    puts 'Welcome to my School Library App'
+    until list_options
+      input = gets.chomp
+      if input == '7'
+        puts 'Thanks for using School Library App (^_^)'
+        break
+      end
+      option input
+    end
+  end
+
+  def list_options
+    puts('-------------------------')
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals of given person id'
+    puts '7 - Quit'
+    puts('-------------------------')
+    print 'Enter number of the operation: '
+  end
+
+  def option(input)
+    case input
+    when '1'
+      list_books
+    when '2'
+      list_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals
+    else
+      'Enter digit from 1 to 7'
+    end
+  end
+
+  def list_books
+    if @books.length.zero?
+      puts 'No Books Available'
+    else
+      @books.each_with_index { |book, idx| puts "#{idx})  Book: #{book.title}, Author: #{book.author}" }
+    end
+    back_to_menu
+  end
+
+  def create_book
+    print 'Enter Book Title: '
+    title = gets.chomp
+    print 'Enter Book Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book)
+    puts 'Book Successfully Created'
+    back_to_menu
+  end
+
+  def create_person
+    puts '1 - Create Student  2 - Create Teacher'
+    select_option
+    person = gets.chomp
+    print 'Insert age: '
+    age = gets.chomp
+    print 'Insert name: '
+    name = gets.chomp
+    person_condition(person, age, name)
+    puts 'Person Created Successfully'
+    back_to_menu
+  end
+
+  def person_condition(person, age, name)
+    case person
+    when '1'
+      print 'Has parents permission [Y/N]: '
+      permit = gets.chomp.downcase
+      parent_permission = permit != 'n'
+      @people.push(Student.new(age, parent_permission, name))
+    when '2'
+      print 'Insert specialization: '
+      specialization = gets.chomp
+      teacher = Teacher.new(specialization, age, name)
+      @people.push(teacher)
+    end
+  end
+
+  def list_people
+    puts 'No person available' if @people.length.zero?
+    @people.each { |person| puts "Name: #{person.name} Age: #{person.age}" }
+    back_to_menu
```

Comment:
- [ ] Great work done with your console app. Nice 👏. There is just one more significant feature for you to implement. If you watch the video, when you input `2`, it should show all the people created with the type of class, i.e either a `Student` or `Teacher`. The output should look like this:

![image](https://user-images.githubusercontent.com/77106962/162550699-d9ac4e50-fd78-4273-963c-79c8e3edb8ae.png)

Kindly implement this feature. 👍

Reply:
Thanks i would implement this feature

Reply:
I am glad you are going to implement it.

Reply:
I have implemented this feature :+1: